### ISSUE TITLE
Standalone output does not load runtimeConfig from environment variables

### ISSUE THREAD

**Issue by blurrah** [vercel/next.js#38119]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64
    Binaries:
      Node: 18.2.0
      npm: 8.9.0
      Yarn: 1.22.18
      pnpm: 7.3.0
    Relevant packages:
      next: 12.2.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### How are you deploying your application? (if relevant)

Docker with `node .next/standalone/server.js`

### Describe the Bug

When running the standalone server using `node .next/standalone/server.js` the server does not load any of the runtimeConfig values from the environment variables, hardcoded values do work. It does get loaded when running the server using `next start`. 

Setting those environment variables during build time does show them when running the server, but that kind of beats the point of using the runtimeConfig and might make for really confusing interactions (it took me for a run).

### Expected Behavior

That running `node .next/standalone/server.js` will load the runtimeConfig variables that are set using an environment variable during runtime.

### Link to reproduction

https://github.com/blurrah/next-runtime-standalone-bug

### To Reproduce

- Build the server using `yarn build`
- Run the server using `TEST_VALUE=test node .next/standalone/server.js`
- You will just see a dash returned on the homepage

When running the server using `TEST_VALUE=test yarn start` it will work properly and return `test - test`.

Using `TEST_VALUE=test yarn build` -> `node .next/standalone/server.js` also returns `test - test`.

**Comment by royvou** [comment]
The issue seems that when the server.js is generated
https://github.com/vercel/next.js/blob/canary/packages/next/build/utils.ts#L1212

It uses the JSON.stringify() on the parsed next.config.js, so all functions are evaluated to static values and they cannot be changed at runtime anymore.

**Comment by durdo** [comment]
I ran into this and would like to understand where we stand on it:

1) Shouldn't this behavior be documented in the `output: "standalone"` section of the docs?
2) Is there a plan to fix this behavior for future releases?
3) Is there and custom server I can use as an example to solve this?

**Comment by mpash** [comment]
Agreed this should be documented. Thanks all for your insight - this helped get my image back on track!

**Comment by ShadiBlitz** [comment]
Is there a current workaround to solve this issue ?

**Comment by ShadiBlitz** [comment]
Why this [issue](https://github.com/vercel/next.js/issues/37537) was closed also ? It's describing almost the same issue ...

**Comment by ShadiBlitz** [comment]
@ijjk 
How can I pass `env` as props if I use it in a regular ts file?

For example

_utils/endpoints.ts_

```
const backendUrl = process.env.BACKEND_URL
console.log(backendUrl)
```
this approach still returns _undefined_. On serverside it works.
I have tried with the prefex `NEXT_PUBLIC_` and still didn't work.

Can you please give an example of how to dynamically use `env` variables from docker run command to standalone build ? I can't seem to follow the above explanation. Thanks!

**Comment by ijjk** [comment]
The above env usage should work with `NEXT_PUBLIC_` and remove the need for passing around the values. Are the `NEXT_PUBLIC_` env variables being set in your environment during build time?
